### DISCUSSION TITLE
Getting involved / contributing to v4

### DISCUSSION THREAD

**Discussion by sastraxi** [nextauthjs/next-auth#2494] [Help]
Hello next-auth team / @balazsorban44,

I see there are a bunch of open issues / discussions / pain points that are coming together for v4. I was wondering how I can help contribute towards this effort? Specifically, I'm interested in helping out with [this issue](https://github.com/nextauthjs/next-auth/pull/2228) as it's something I'm running up against getting my app production-ready while hosted on vercel.

I'm also open to helping out with documentation, test coverage, or anything else that might help the effort out. I'm posting here in Discussions as I didn't see any (e.g.) discord server for this kind of coordination.

**Comment by balazsorban44** [comment]
Hi, and thank you! Yes I generally don't like Discord/Slack as they are hidden from those not using them and it's not indexed by search engines either.

V4 is getting in shape, but there are still a few things left, which we are actually going to discuss today with the others.

One recurring problem that takes up most of my time is triaging issues. It takes a bit of knowledge of the library, but many questions can be answered by other users who has a good understanding of OAuth and our library's goal. Answering questions/issues, asking people for  reproduction/testing those reproductions is something I would need help with any time of the day. 😅

Furthermore, we just introduced a new way of configuration for OAuth 2 and OIDC providers, but we haven't tested/refactored all the built-in ones yet. (see #2411). I would like to get as many OAuth provider accounts as possible, to make this painless for people when v4 hits stable. See the PR and the related RFC to better understand what is going on.

#2361 is my last bigger PR I would like to finish. When that's done, we will have to refactor the Adapters like this: https://github.com/nextauthjs/adapters/pull/171

Using that PR, I would appreciate help with converting the other adapters as well.

We will have to update our `next-auth-example` and `next-auth-typescript-example` repositories also.

For the particular issue you want to help out with, it's actually a fairly simple refactor (I do have a plan already), but currently it's a bit risky as we don't have TypesScript and so any core refactoring makes me a little bit nervous. Unfortunately, I don't think it will make it to the first v4 release, but it can be one of the first features added maybe.

Once things fall into place for v4, I would like to start converting the core to TypeScript, probably gradually through multiple PRs. If you want to help out with that, that would be awesome. 😊 I know for a fact that @lluia is interested to get this done.

**Reply by sastraxi** [reply]
I've subscribed to #2361 so I know when it lands; happy to help migrate the other adapters over! It looks like you and the team are bringing more and more typescript in already which is great to see.

So my plan will be to try to get acquainted with the `next` codebase and prepare to help move over adapters. When I feel comfortable enough with the codebase I'll start looking at issues coming in and try to help reduce the noise so that the team can focus on v4.

**Reply by balazsorban44** [reply]
That is immensely appreciated! :pray: :green_heart: Feel free to reach out if you have any questions!
